architecture/libraries.rst

```diff
@@ -4,24 +4,88 @@
 Libraries
 #########
 
+.. _libutils:
+
+libutils
+*******
+
+OP-TEE core and OP-TEE development kit for Trusted Application provide a
+libc standard library that is named **libtuils**. It implements many
+standard functions like ``snprintf()``, ``strncmp()``, ``memcpy()``,
+``malloc()``. ``qsort()``, and many more but not all standard libc functions.
+
+Note however that Trusted Applications implemented in "C" should use GP TEE
+Internal Core API functions rather than their stdlib equivalent. This makes
+those TAs implementation more portable to other GP TEE compliant environments.
+
+When ``CFG_ULIBS_SHARED`` is enabled, **libutils** is assigned UUID
+**71855bba-6055-4293-a63f-b0963a737360**.
+
 .. _libutee:
 
 libutee
 *******
 The :ref:`tee_internal_core_api` describes services that are provided to Trusted
 Applications. **libutee** is a library that implements this API.
 
-libutee is a static library the Trusted Applications shall statically link
-against. Trusted Applications do execute in non-privileged secure userspace and
-libutee also aims at being executed in the non-privileged secure userspace.
+libutee is designed as a userland library specifically dedicated to OP-TEE
+Trusted Applications and aims at being executed in the non-privileged secure
+userspace.
 
 Some services for this API are fully statically implemented inside the libutee
 library while some services for the API are implemented inside the OP-TEE core
 (privileged level) and libutee calls such services through system calls.
 
-.. _libmpa:
+When ``CFG_ULIBS_SHARED`` is enabled, **libutee** is assigned UUID
+**4b3d937e-d57e-418b-8673-1c04f2420226**.
 
-libmpa
+libmbedtls
+**********
+
+OP-TEE OS source tree provides support of the mbed TLS library, named
+**libmbedtls**.
+
+A specific build sequence compiles an instance of **libmbedtls** and links
+it to OP-TEE. Another build sequence compiles an instance of **libmbedtls**
```

Comment:
"links it to OP-TEE": only when `CFG_CRYPTOLIB_NAME=mbedtls CFG_CRYPTOLIB_DIR=core/lib/libtomcrypt`, otherwise libtomcrypt is used.